webapp/channels/src/components/drafts/drafts.tsx

```diff
@@ -39,14 +40,29 @@ function Drafts({
     const dispatch = useDispatch();
     const {formatMessage} = useIntl();
 
+    const history = useHistory();
+    const match: match<{team: string}> = useRouteMatch();
+    const isDraftsTab = useRouteMatch('/:team/drafts');
+    const isScheduledPostsTab = useRouteMatch('/:team/scheduled_posts');
+
     useEffect(() => {
         dispatch(selectLhsItem(LhsItemType.Page, LhsPage.Drafts));
         dispatch(suppressRHS);
 
         return () => {
             dispatch(unsuppressRHS);
         };
-    }, []);
+    }, [dispatch]);
+
+    const handleSwitchTabs = useCallback((key) => {
+        if (key === 0 && isScheduledPostsTab) {
```

Comment:
Why are we checking `isScheduledPostsTab` here? Couldn't we just do this based on the `key`?

Reply:
You don't wanna navigate from scheduled post tab URL again to the same URL, adding duplicate items in history. Now checking for this would mean if you click on scheduled post tab 5 times, going back in browser will keep you at the same page for 5 times. Checking beyond the key prevents that.

Reply:
Ah, I see. I would've expected that it wouldn't call this at all if you're already on the corresponding tab, but I guess not